### ISSUE TITLE
Phase 2.2: Add EQ_predict inference-only entry point

### ISSUE THREAD

**Issue by mmcdermott** [payalchandak/EveryQuery#81]
Parent: #54 (Phase 2.2). Depends on: #80 (schema), #79 (submodule restructure).

## Goal

Add an inference-only `EQ_predict` entry point that takes `(trained_model_run_dir, tasks.parquet)` and writes prediction parquets. Fully decoupled from metrics, model selection, or AUC computation.

## Why split this out

Today `EQ_evaluate` conflates four concerns: index-time sampling, task-matrix slicing, inference, and metric computation. Separating inference lets external consumers run the model on their own tasks without inheriting our metric conventions, and gives `EQ_evaluate` a clean input (predictions) it can compute metrics over.

## Scope

- New submodule: `src/every_query/predict/`
- New Hydra entry point: `EQ_predict`
- Input: `(model_run_dir, tasks_parquet_matching_schema)` where the schema is whatever #80 lands.
- Output: prediction parquets. Shape: the input rows plus a `predicted_boolean_probability` column (name TBD to match MEDS / meds-evaluation conventions).
- One model per invocation. Resolves the "why a list of model_run_dirs?" question from #66 by making the list go away.

## Explicitly excluded

- No AUCs. No ROC/PR. No per-bucket aggregation. No win-rate ranking. No model selection.
- No ID/OOD splitting. The caller provides the tasks parquet.
- No checkpoint selection policy — `model_run_dir` is expected to already resolve to one checkpoint (e.g. `best_model.ckpt`).

## Open questions

- Batch / streaming write — does prediction write one-parquet-per-shard, one-parquet-total, or follow the input's layout?
- Does `EQ_predict` load the model's own training config from `resolved_config.yaml` to recover datamodule settings, or does it accept an explicit datamodule config? (First is nicer, second is more flexible.)
- How do we handle codes present in the tasks parquet but absent from the model's training vocabulary? Error, warn, or skip?

## Acceptance criteria

- [ ] `EQ_predict --help` works after `pip install`.
- [ ] Given a trained demo model + a tasks parquet, produces a predictions parquet with the agreed schema.
- [ ] CPU-only integration test in `tests/test_predict_cli.py` that exercises the full subprocess path.

## Dependencies

- #80 (schema) must be decided before this starts implementation. Until then this issue is a placeholder.
- #79 (submodule restructure) — `EQ_predict` will live under `predict/` alongside the restructured `prepare_tasks/` and `pretrain/`.

**Comment by mmcdermott** [comment]
## Design doc — to sign off before implementation

Drafting this now that PR #96 defines `TaskQuerySchema`.  Answers the open questions from the issue body + sketches the implementation so we can handshake before writing code.

### Proposed I/O

**Inputs** (CLI via Hydra, single invocation = single model):
1. `model_run_dir` — path to a training run dir with `best_model.ckpt` + `resolved_config.yaml` (the pattern `evaluate/eval.py::_setup_model` already reads).
2. `tasks_parquet` — a parquet file conformant to `TaskQuerySchema` (#96).  Each row: `(subject_id, prediction_time, code, duration_days)` with optional inherited label columns (ignored by predict, consumed by evaluate).

**Output**: a predictions parquet conformant to a new `PredictionSchema(TaskQuerySchema)` that adds one column:
- `predicted_boolean_probability: Optional(float32)` — the model's `sigmoid(logit)` for "code will occur within duration_days of prediction_time."

Matches meds-evaluation's `PredictionSchema` pattern but extending EQ's `TaskQuerySchema` instead of the bare `LabelSchema`.

### Proposed submodule layout

```
src/every_query/predict/
├── __init__.py          (already exists from #90)
├── predict.py           NEW — the EQ_predict Hydra main
├── schema.py            NEW — PredictionSchema(TaskQuerySchema)
├── configs/
│   └── predict.yaml     NEW — required: model_run_dir, tasks_parquet, output_parquet
├── external_tasks/      (already exists from #90)
└── README.md            (already exists; extend)
```

### Proposed algorithm for `predict.py::main`

```
1. Load tasks_parquet via pa.parquet → validate with TaskQuerySchema.align(table).
2. _setup_model(model_run_dir)  [ported from evaluate/eval.py, unchanged]
3. Group the tasks_df by `code` (since the current EveryQueryPytorchDataset indexes per-code
   task-labels dirs). For each (code, slice_of_tasks):
   a. Materialize the slice as a per-code task_labels parquet in a tmpdir.
   b. train_cfg.datamodule.config.task_labels_dir = tmpdir
   c. D = instantiate(train_cfg.datamodule)
   d. preds = trainer.predict(model=M, datamodule=D, ckpt_path=None)
   e. Extract (subject_id, prediction_time, occurs_probs) from the batches.
   f. Append to the running output.
4. Join the collected predictions back to the original tasks_df rows so output ordering +
   (code, duration_days) carry through.
5. Write output_parquet as PredictionSchema-conformant.
```

The group-by-code step is load-bearing: the existing dataset expects a single-code task-labels dir. A future #TODO could teach the dataset to ingest a multi-code parquet directly (would avoid the tmpdir roundtrip), but that's optimization, not correctness.

### Explicitly out of scope for the initial PR

- **No AUCs, no metrics, no model selection.**  Those live in `EQ_evaluate` (#83) on top of this PR's output.
- **No multi-model loop.**  `model_run_dirs` (plural) was in the old `eval.py::main`; here it's singular.  Cross-model comparison is paper-experiments (#54 plan).
- **No ID/OOD bucket labeling on the output.**  PredictionSchema carries only the prediction probability; bucket assignment is an evaluation/paper-experiments concern.
- **No new query types.**  Flat `(code, duration_days)` only, per the TaskQuerySchema decision on #80.

### Open questions for sign-off

1. **Output naming.**  `predicted_boolean_probability` matches meds-evaluation.  Alternative: `prob_occurs` — more EQ-native but breaks symmetry with meds-evaluation.  Recommendation: stick with `predicted_boolean_probability` for interop.
2. **Should predict also output query embeddings?**  The current `evaluate/eval.py::_run_predict` writes both `eval_preds_*.parquet` AND `eval_embeds_*.parquet`.  Two options:
   - **(a)** Keep embeddings as a separate optional output parquet with its own schema.
   - **(b)** Drop embeddings from `EQ_predict` — they're a paper-experiments-figure concern, not a normal-usage output.
   - Recommendation: **(a)** with an opt-in flag (`--write-embeddings`) — keeps the existing capability available without making it the default.
3. **How strict should we be about the inherited label columns?**  TaskQuerySchema inherits `boolean_value` etc. as Optional.  If present on the input, should `EQ_predict` carry them through to the output untouched?  (Yes — lets downstream evaluate just join on `(subject_id, prediction_time, code, duration_days)`.)
4. **Error handling when `code` isn't in the model's training vocabulary.**  Skip with a warning, or error out?  Recommendation: warn + skip (same pattern `_run_predict` uses today for missing task_labels_dir).

### After sign-off

Plan: one PR doing steps 1-5 above.  ~200 lines of code + one new schema file + one Hydra config + README update + a `tests/test_predict_cli.py` with a subprocess integration test.  Closes #81 on merge.

**Comment by mmcdermott** [comment]
Shipped in #99 — `EQ_predict` inference-only entry point.  Takes a `model_run_dir` + a `tasks_dir` of `TaskQuerySchema`-conformant parquets, writes a `PredictionSchema`-conformant parquet (subject_id, prediction_time, query, duration_days, boolean_value, censor_prob, occurs_prob).  Single order-preserving pass, schema-driven names, subprocess-tested end-to-end (including the slow training-validity E2E which now runs predictions through EQ_predict).  Follow-up: MTD upstream issue to let `task_labels_dir` accept file / glob inputs — draft content ready, sandbox blocked the external-repo write.
